crates/milli/src/filterable_attributes_rules.rs

```diff
@@ -0,0 +1,204 @@
+use deserr::{DeserializeError, Deserr, ValuePointerRef};
+use serde::{Deserialize, Serialize};
+use std::collections::{BTreeSet, HashSet};
+use utoipa::ToSchema;
+
+use crate::{
+    attribute_patterns::{match_distinct_field, match_field_legacy, PatternMatch},
+    constants::RESERVED_GEO_FIELD_NAME,
+    AttributePatterns, FieldsIdsMap,
+};
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Debug, ToSchema)]
+#[serde(untagged)]
+pub enum FilterableAttributesRule {
+    Field(String),
+    Pattern(FilterableAttributesPatterns),
+}
+
+impl FilterableAttributesRule {
+    pub fn match_str(&self, field: &str) -> PatternMatch {
+        match self {
+            FilterableAttributesRule::Field(pattern) => match_field_legacy(pattern, field),
+            FilterableAttributesRule::Pattern(patterns) => patterns.match_str(field),
+        }
+    }
+
+    pub fn has_geo(&self) -> bool {
+        matches!(self, FilterableAttributesRule::Field(field_name) if field_name == RESERVED_GEO_FIELD_NAME)
```

Comment:
This means that only `"_geo"` as its string from is allowed, right?

In particular, the PRD has an example with:
```jsonc
{
    "filterableAttributes": [
	    // Good way
	    {
	      "attributePatterns": ["_geo"]
	    }
    ]
}
``` 

But reading the code it appears that only the following would work:

```jsonc
{
    "filterableAttributes": [ "_geo" ]
}
```

Reply:
let's modify the doc to fit the current behavior, enhance the doc